application/ui/src/features/inspect/dataset/dataset-status-panel.component.tsx

```diff
@@ -168,12 +172,60 @@ const TrainingInProgress = ({ job }: TrainingInProgressProps) => {
     return null;
 };
 
+const REFETCH_INTERVAL_WITH_TRAINING = 1_000;
+
+const useProjectTrainingJobs = () => {
+    const queryClient = useQueryClient();
+    const { projectId } = useProjectIdentifier();
+    const prevJobsRef = useRef<Job[]>([]);
+
+    const { data } = $api.useQuery('get', '/api/jobs', undefined, {
+        refetchInterval: ({ state }) => {
+            const projectHasTrainingJob = state.data?.jobs.some(
+                ({ project_id, type, status }) =>
+                    projectId === project_id && type === 'training' && (status === 'running' || status === 'pending')
+            );
+
+            return projectHasTrainingJob ? REFETCH_INTERVAL_WITH_TRAINING : undefined;
+        },
+    });
+
+    useEffect(() => {
```

Comment:
Perhaps extract this into a named hook like `useRefreshModelsOnJobUpdates`? (because it took me a while to realize what it was doing)